cmd/mattermost/commands/team.go

```diff
@@ -285,6 +302,54 @@ func searchTeamCmdF(command *cobra.Command, args []string) error {
 	return nil
 }
 
+func archiveTeamsCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Enter at least one team to archive.")
+	}
+	teams := getTeamsFromTeamArgs(a, args)
+	for i, team := range teams {
+		if team == nil {
+			CommandPrintErrorln("Unable to find team '" + args[i] + "'")
+			continue
+		}
+		if result := <-a.Srv.Store.Team().Delete(team.Id, model.GetMillis()); result.Err != nil {
```

Comment:
We already have implemented the delete in `app/team.go` as `SoftDeleteTeam`. I would use that. And implement a `RestoreTeam` function in `app/team.go` to do the inverse process.